graphql_test.go

```diff
@@ -4113,7 +4195,9 @@ func TestTracer(t *testing.T) {
 		"id": "1002",
 	}
 
-	_ = schema.Exec(ctx, doc, opName, variables)
+	visitors := map[string]types.DirectiveVisitor{}
+
+	_ = schema.Exec(ctx, doc, opName, variables, visitors)
```

Comment:
I do believe that directive visitors should be passed at schema parse time and need to be a member of the schema itself and not members of the requests. IMHO, the resolvers do not need to be a member of the `Request` struct. and here you have access to the schema too.